modules/sor/sorV2/lib/poolsV2/gyroE/gyroEPool.ts

```diff
@@ -178,6 +181,8 @@ export class GyroEPool implements BasePool {
         mutateBalances?: boolean,
     ): TokenAmount {
         const { tIn, tOut } = this.getPoolTokens(tokenIn, tokenOut);
+        // swapAmount from PathWithAmount does not take rate into account - this step fixes scale18 so it properly considers rates
+        const swapAmountLive = new PoolTokenWithRate(tokenOut, swapAmount.amount, tOut.index, tOut.rate);
```

Comment:
This was the main fix. It makes sure rates will be properly applied to `swapAmount`